tests/integration/conftest.py

```diff
@@ -143,6 +157,144 @@ def iperf3_pods(client):
+
+    with ops_test.model_context("main"):
+        log.info(f"Removing cloud '{cloud_name}'...")
+        await ops_test.juju(
+            "remove-cloud",
+            cloud_name,
+            "--controller",
+            ops_test.controller_name,
+            "--client",
+            check=True,
+        )
+
+
+@pytest.fixture(scope="module")
+async def k8s_model(k8s_cloud, ops_test):
+    model_alias = "k8s-model"
+    log.info("Creating k8s model ...")
+    # Create model with Juju CLI to work around a python-libjuju bug
+    # https://github.com/juju/python-libjuju/issues/603
+    model_name = "test-kube-ovn-" + "".join(choices(ascii_lowercase + digits, k=4))
+    await ops_test.juju(
+        "add-model",
+        f"--controller={ops_test.controller_name}",
+        model_name,
+        k8s_cloud,
+        "--no-switch",
+    )
+    model = await ops_test.track_model(
+        model_alias,
+        model_name=model_name,
+        cloud_name=k8s_cloud,
+        credential_name=k8s_cloud,
+        keep=False,
+    )
+    model_uuid = model.info.uuid
+    yield model, model_alias
+    timeout = 5 * 60
+    await ops_test.forget_model(model_alias, timeout=timeout, allow_failure=False)
+
+    async def model_removed():
+        _, stdout, stderr = await ops_test.juju("models", "--format", "yaml")
+        if _ != 0:
+            return False
+        model_list = yaml.safe_load(stdout)["models"]
+        which = [m for m in model_list if m["model-uuid"] == model_uuid]
+        return len(which) == 0
+
+    log.info("Removing k8s model")
+    await juju.utils.block_until_with_coroutine(model_removed, timeout=timeout)
+    # Update client's model cache
+    await ops_test.juju("models")
+    log.info("k8s model removed")
+
+
+@pytest.fixture(scope="module")
+async def multus_installed(ops_test, k8s_model):
+    _, k8s_alias = k8s_model
+    with ops_test.model_context(k8s_alias) as model:
+        await model.deploy(entity_url="multus", channel="edge")
+        await model.block_until(lambda: "multus" in model.applications, timeout=60)
+        await model.wait_for_idle(status="active", timeout=60 * 60)
+
+    # need to wait until all kubernetes-worker units have multus CNI config installed
+    deadline = time.time() + 600
+    for unit in ops_test.model.applications["kubernetes-worker"].units:
+        log.info("waiting for Multus config on unit %s" % unit.name)
+        while time.time() < deadline:
+            rc, _, _ = await ops_test.juju(
+                "ssh",
+                "-m",
+                ops_test.model_full_name,
+                unit.name,
+                "--",
+                "sudo",
+                "ls",
+                "/etc/cni/net.d",
+                "|",
+                "grep",
+                "multus",
+            )
+            if rc == 0:
+                break
+            await asyncio.sleep(1)
+        else:
+            pytest.fail("timed out waiting for Multus config on unit %s" % unit.name)
+
```

Comment:
You might be running into issues with pytest not being able to remove the model because of the storage. I ran into that in my grafana tests. I was able to get passed the block by removing the application like so:
```
    # Deploy multus app ...
    ...
   yield
    with ops_test.model_context(k8s_alias) as m:
        log.info("Removing grafana-k8s application ...")
        cmd = "remove-application multus --destroy-storage --force"
        rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))

Reply:
Thanks. I'm not running into that, the model does tear down cleanly. I think that's because the Multus charm doesn't have any storage defined in its metadata, so there's no Juju modeled storage to destroy.

Reply:
yeah looks like yours tears down fine, cool!